packages/eddsa-pcd/src/EdDSAPCD.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  DisplayOptions,
+  PCD,
+  PCDPackage,
+  SerializedPCD,
+  StringArgument,
+  StringArrayArgument
+} from "@pcd/pcd-types";
+import { buildEddsa, buildPoseidon } from "circomlibjs";
+import { v4 as uuid } from "uuid";
+import { EdDSACardBody } from "./CardBody";
+
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference
+/// <reference path="util/declarations/circomlibjs.d.ts" />
+
+export const EdDSAPCDTypeName = "eddsa-pcd";
+
+export interface EdDSAPCDArgs {
+  privateKey: StringArgument;
+  message: StringArrayArgument;
+  id: StringArgument;
```

Comment:
why do we need `id`? I see it also in rsa-pcd but i don't see it in any other pcds

Reply:
cc @ichub

Reply:
All PCDs have an id. most implementations generate a new `uuid` for each instance of a pcd. some pcds let you specify an id in the arguments. the id of the pcd is typically separate from the content of a pcd. 

a related thing is you have an issue (#402 ) out to make pcd ids be content hashes. i think it's a neat idea but not sure it totally makes sense. eg. in file systems you're able to have duplicate file contents with different names.